qiskit/compiler/assembler.py

```diff
@@ -142,40 +142,31 @@ def assemble_schedules(schedules, dict_config, dict_header):
     Raises:
         QiskitError: when invalid command is provided
     """
-    if isinstance(schedules, Schedule):
+    if isinstance(schedules, ConfiguredSchedule):
         schedules = [schedules]
 
     experiments = []
 
-    for ii, schedule in enumerate(schedules):
-
-        # use LO frequency configs
-        default_qubit_lo_freq = dict_config.get('qubit_lo_freq', None)
-        default_meas_lo_freq = dict_config.get('meas_lo_freq', None)
-
-        # TODO: scheudle has no information of LO frequency
-        # lo_freqs = {}
-        # user_qubit_lo_freq = [q.drive.lo_frequency for q in schedule.device.q]
-        # if default_qubit_lo_freq:
-        #     if default_qubit_lo_freq != user_qubit_lo_freq:
-        #         lo_freqs['qubit_lo_freq'] = user_qubit_lo_freq
-        # else:
-        #     lo_freqs['qubit_lo_freq'] = user_qubit_lo_freq
-        # user_meas_lo_freq = [q.measure.lo_frequency for q in schedule.device.q]
-        # if default_meas_lo_freq:
-        #     if default_meas_lo_freq != user_meas_lo_freq:
-        #         lo_freqs['meas_lo_freq'] = user_meas_lo_freq
-        # else:
-        #     lo_freqs['meas_lo_freq'] = user_meas_lo_freq
+    # use LO frequency configs
+    default_qubit_lo_freq = dict_config.get('qubit_lo_freq', None)
+    default_meas_lo_freq = dict_config.get('meas_lo_freq', None)
+
+    user_pulselib = set()
+    for ii, configed in enumerate(schedules):
+        schedule = configed.schedule
+
+        lo_freqs = {
```

Comment:
Please avoid adding them if default and user_configs are the same. Unnecessary header configs just increase data size.

Reply:
OK. In the case of default_los == user_los, we need to set `lo_freqs = {}`, right?

Reply:
done!